test/types/models.test.ts

```diff
@@ -977,3 +978,29 @@ function testWithLevel1NestedPaths() {
     'foo.one': string | null | undefined
   }>({} as Test2);
 }
+
+function gh14764TestFilterQueryRestrictions() {
+  const TestModel = model<{ validKey: number }>('Test', new Schema({}));
+  // @ts-expect-error: A key not in the schema should be invalid
+  TestModel.find({ invalidKey: 0 });
+  // @ts-expect-error: A key not in the schema should be invalid for simple root operators
+  TestModel.find({ $and: [{ invalidKey: 0 }] });
+
+  // Any "nested" keys should be valid
+  TestModel.find({ 'validKey.subkey': 0 });
+
+  // And deeply "nested" keys should be valid
+  TestModel.find({ 'validKey.deep.nested.key': 0 });
+  TestModel.find({ validKey: { deep: { nested: { key: 0 } } } });
+
+  // Any Query should be accepted as the root argument (due to merge support)
+  TestModel.find(TestModel.find());
+  // @ts-expect-error: A Query should not be a valid type for a FilterQuery within an op like $and
+  TestModel.find({ $and: [TestModel.find()] });
+
+  const id = new Types.ObjectId();
+  // Any ObjectId should be accepted as the root argument
+  TestModel.find(id);
+  // @ts-expect-error: A ObjectId should not be a valid type for a FilterQuery within an op like $and
+  TestModel.find({ $and: [id] });
+}
```

Comment:
this is not how the type errors are tested, look at other tests.
instead of `@ts-expect-error` comments [`expectError`](https://github.com/Automattic/mongoose/blob/ea2a4ef7b20f73217b82dd21068a2f85bc9b197d/test/types/models.test.ts#L257) should be used (even if your IDE says there is a error)